helpers.h

```diff
@@ -316,8 +316,13 @@ get_edges_from_file_mtx(const std::string &filename, uint64_t *edge_count,
   int64_t c = strtol(W.Strings[6], nullptr, 10);
   //{parallel_for(uint64_t i=0; i < len; i++) In[i] = atol(W.Strings[i + 1]);}
 
-  if (W.m - 8 != m * 3) {
-    std::cout << "Bad input file: length = " << W.m - 8 << " 3m = " << 3 * m
+  uint32_t elements_per_line = 3;
+  if (pattern) {
+    elements_per_line = 2;
+  }
+
+  if (W.m - 8 != m * elements_per_line) {
+    std::cout << "Bad input file: length = " << W.m - 8 << elements_per_line << "m = " << elements_per_line * m
```

Comment:
I think you want some commas between the output elements here, also make sure that the thing you are printing out is the same as the thing you say you are printing out.

Reply:
Added some commas and also print out the elements per line variable.  See example message below.

Bad input file: length = 39, elements per line 2, m = 40
%%MatrixMarket, matrix, coordinate, pattern, symmetric, 20, 20, 20, 2